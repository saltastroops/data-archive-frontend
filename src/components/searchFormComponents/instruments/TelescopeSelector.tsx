import * as React from "react";
import MultiSelectField, {
  AllOption
} from "../../basicComponents/MultiSelectField";

interface ITelescopeSelectorProps {
  telescopes?: string[];
  onSelect: (value: any) => void;
}

// const TELESCOPES = ["SALT", "1.9 m", "Lesedi"];
const TELESCOPES = ["SALT"];
/**
 * A form for selecting telescope names.
 */
const TelescopeSelector = (props: ITelescopeSelectorProps) => {
  const { onSelect, telescopes } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      telescopes: Array.from(values).map((t: any) => {
        return t.text;
      })
    });
  };

  return (
    <div>
      <p>Telescope</p>
      <MultiSelectField
        name={"telescopes"}
        onChange={select}
        value={telescopes || ["All"]}
      >
        <AllOption />
        {TELESCOPES.sort().map(i => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default TelescopeSelector;
