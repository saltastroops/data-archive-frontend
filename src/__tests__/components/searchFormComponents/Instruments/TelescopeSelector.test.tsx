import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import TelescopeSelector from "../../../../components/searchFormComponents/instruments/TelescopeSelector";

const onSelect = jest.fn() as any;
describe("TelescopeSelector ", () => {
  it("should be defined", () => {
    expect(
      mount(<TelescopeSelector telescopes={[]} onSelect={onSelect} />)
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(mount(<TelescopeSelector telescopes={[]} onSelect={onSelect} />))
    ).toMatchSnapshot();
  });
});
