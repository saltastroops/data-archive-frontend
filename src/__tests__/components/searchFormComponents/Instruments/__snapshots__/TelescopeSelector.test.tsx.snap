// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelescopeSelector  should render correctly 1`] = `
.c0 {
  width: 95%;
}

<TelescopeSelector
  onSelect={[MockFunction]}
  telescopes={Array []}
>
  <div>
    <p>
      Telescope
    </p>
    <MultiSelectField
      name="telescopes"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <div
          className="c0 control select is-multiple "
        >
          <select
            multiple={true}
            name="telescopes"
            onChange={[Function]}
            size={6}
            style={
              Object {
                "width": "216px",
              }
            }
            value={Array []}
          >
            <AllOption>
              <option
                value="All"
              >
                All
              </option>
            </AllOption>
            <option
              key="SALT"
              value="SALT"
            >
              SALT
            </option>
          </select>
        </div>
      </styled.div>
    </MultiSelectField>
  </div>
</TelescopeSelector>
`;
